server/utils/cloudFunctions/functions/test/algoliaTests/algoliaIntegrationTests.test.js

```diff
@@ -0,0 +1,237 @@
+/**
+ * Description: Integration tests for algolia cloud functions.
+ *              Can be flaky. If an error like `ObjectID` does not exist, 
+ *              it is probably a problem with algolia flaking. When in doubt,
+ *              re-run the tests. Compile with `npm run test`.
+ * Notes:
+ * - For simplicity, advertisment json's are denoted as `ad`, and 
+ *   the advertiser is used as the documents `primary key`. 
+ *   - In practice, primary key is a random string.
+ *   - If used something like `adOne` as `primary key`, would probably
+ *     run into a lot of collisions with other `adOne`'s not deleted
+ *     fast enough between tests (see note about `after` cleanup below.)
+ *   - Advertiser naming convention for this test suite is alphabetical 
+ *     enumeration. This is in contrast to the `countAdvertisersFunction`
+ *     test suite, which used alphabetical `ad_x` enumeration. 
```

Comment:
Are lengthy notes like this supposed to go in the file? I'm not sure of the convention but it seems like it might be better placed in a different file. Not sure though, maybe something to check w/ the hosts.